test/JDBC/expected/temp_table_rollback-vu-verify.out

```diff
@@ -0,0 +1,386 @@
+
+
+-- Duplicate key doesn't cause implicit rollback, so the drop will succeed here. 
+BEGIN TRAN
+drop table #temp_table_rollback_t2
+insert into #temp_table_rollback_t1 values (1, 1, 'a')
+insert into #temp_table_rollback_t1 values (1, 1, 'a')
+GO
+~~ROW COUNT: 1~~
+
+~~ERROR (Code: 2627)~~
+
+~~ERROR (Message: duplicate key value violates unique constraint "#temp_table_rollback_t1_pkey")~~
+
+
+SELECT * FROM #temp_table_rollback_t2
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: relation "#temp_table_rollback_t2" does not exist)~~
+
+
+BEGIN TRAN
+DROP TABLE #temp_table_rollback_t1
+DROP TABLE #temp_table_rollback_t2
+COMMIT
+GO
+
+----------------------------------------------------------
+-- Nested transactions
+----------------------------------------------------------
+BEGIN TRANSACTION T1
+    CREATE TABLE #t1(a varchar(16))
+    INSERT INTO #t1 VALUES ('t1')
+    SAVE TRANSACTION T2
+        INSERT INTO #t1 VALUES ('t2')
+        SAVE TRANSACTION T3
+            INSERT INTO #t1 VALUES ('t3')
+            SAVE TRANSACTION T4
+                INSERT INTO #t1 VALUES ('t4')
+                SELECT * FROM #t1
+            ROLLBACK TRANSACTION T4
+            SELECT * FROM #t1
+        ROLLBACK TRANSACTION T3
+        SELECT * FROM #t1
+    ROLLBACK TRANSACTION T2
+    SELECT * FROM #t1
+ROLLBACK TRANSACTION T1
+SELECT * FROM #t1
+GO
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~START~~
+varchar
+t1
+t2
+t3
+t4
+~~END~~
+
+~~START~~
+varchar
+t1
+t2
+t3
+~~END~~
+
+~~START~~
+varchar
+t1
+t2
+~~END~~
+
+~~START~~
+varchar
+t1
+~~END~~
+
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: relation "#t1" does not exist)~~
+
+
+----------------------------------------------------------
+-- Savepoint CREATE and DROP
+----------------------------------------------------------
+BEGIN TRANSACTION T1
+    CREATE TABLE #t1(a int primary key, b varchar(16))
+    SAVE TRANSACTION T2
+        INSERT INTO #t1 VALUES (1, 't2')
+        SAVE TRANSACTION T3
+            UPDATE #t1 SET a = 2 WHERE a = 1
+            SELECT * FROM #t1
+            SAVE TRANSACTION T4
+                DROP TABLE #t1
+            ROLLBACK TRANSACTION T4
+        ROLLBACK TRANSACTION T3
+        SELECT * FROM #t1
+    ROLLBACK TRANSACTION T2
+select * from enr_view
+GO
+~~ROW COUNT: 1~~
+
+~~ROW COUNT: 1~~
+
+~~START~~
+int#!#varchar
+2#!#t2
+~~END~~
+
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: relation "#t1" does not exist)~~
+
+
+
+---------------------------------------------------------------------------
+-- Same temp table name in one transaction
+---------------------------------------------------------------------------
+CREATE TABLE #temp_table_rollback_t3(c1 INT, c2 INT)
+GO
+
+BEGIN TRANSACTION
+    ALTER TABLE #temp_table_rollback_t3 ADD C3 INT;
+    INSERT INTO #temp_table_rollback_t3 VALUES (1, 2, 3)
+    DROP TABLE #temp_table_rollback_t3
+    CREATE TABLE #temp_table_rollback_t3(c1 INT, c2 INT)
+COMMIT
+GO
```

Comment:
A slightly different version of this is crashing as well.

```
CREATE TABLE #temp(c1 INT, c2 CHAR(10))
GO

BEGIN TRANSACTION T1
ALTER TABLE #temp ADD C3 INT
DROP TABLE #temp

CREATE TABLE #temp(c1 INT, c2 CHAR(10))
DROP  TABLE #temp

INSERT INTO #temp VALUES (2, 'two') -- Crashed
COMMIT
GO
```

Reply:
fixed